ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -463,7 +466,44 @@ private TopicDescription getTopicDescription(final Admin admin, final String sou
           // special case where we expect no work at all on the partition, so we don't even
           // need to check the committed offset (if we did, we'd potentially wait forever,
           // since Streams won't commit anything for an empty topic).
-          .filter(e -> e.getValue().offset() > 0L)
+          .filter(e -> e.getValue().offset() > 0L && e.getValue().offset() > startOffsetsForStreamPullQuery.get(e.getKey()))
```

Comment:
Could you augment the above comment as well?

Reply:
Ah, I thought it still applied, but I can see the value of providing more detail.